src/arrayMethodJoin.js

```diff
@@ -5,7 +5,33 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    let newSeparator;
+
+    if (separator === undefined) {
+      newSeparator = ',';
+    } else {
+      newSeparator = separator;
+    }
+
+    if (this.length === 0) {
+      return '';
+    }
```

Comment:
Unnecessary. Loop doesn’t call with an empty array (you added `i < this.length` check)